Content.Server/Xenoarchaeology/XenoArtifacts/ArtifactSystem.cs

```diff
@@ -0,0 +1,70 @@
+using Content.Server.Xenoarchaeology.XenoArtifacts.Events;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Xenoarchaeology.XenoArtifacts;
+
+public class ArtifactSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IComponentFactory _componentFactory = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ArtifactComponent, ComponentInit>(OnInit);
+    }
+
+    private void OnInit(EntityUid uid, ArtifactComponent component, ComponentInit args)
+    {
+        if (component.RandomTrigger)
+        {
+            AddRandomTrigger(uid, component);
+        }
+    }
+
+    public void AddRandomTrigger(EntityUid uid, ArtifactComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+
+        var triggerName = _random.Pick(component.PossibleTriggers);
+        var trigger = (Component) _componentFactory.GetComponent(triggerName);
+        trigger.Owner = uid;
+
+        EntityManager.AddComponent(uid, trigger);
```

Comment:
acruid expressed some concerns over this so may need to get a 2nd opinion but I approve

Reply:
@Acruid

Reply:
This works, I don't think there is any other way currently to add a component to an entity using the component name.